core/src/main/java/jenkins/telemetry/Telemetry.java

```diff
@@ -220,31 +218,26 @@ protected void execute(TaskListener listener) throws IOException, InterruptedExc
                 String correlationId = ExtensionList.lookupSingleton(Correlator.class).getCorrelationId();
                 wrappedData.put("correlator", DigestUtils.sha256Hex(correlationId + telemetry.getId()));
 
+                String body = wrappedData.toString();
+                if (LOGGER.isLoggable(Level.FINEST)) {
+                    LOGGER.finest("Submitting JSON: " + body);
+                }
+                HttpClient httpClient = ProxyConfiguration.newHttpClient();
+                HttpRequest httpRequest;
                 try {
-                    URL url = new URL(ENDPOINT);
-                    URLConnection conn = ProxyConfiguration.open(url);
-                    if (!(conn instanceof HttpURLConnection)) {
-                        LOGGER.config("URL did not result in an HttpURLConnection: " + ENDPOINT);
-                        return;
-                    }
-                    HttpURLConnection http = (HttpURLConnection) conn;
-                    http.setRequestProperty("Content-Type", "application/json; charset=utf-8");
-                    http.setDoOutput(true);
-
-                    String body = wrappedData.toString();
-                    if (LOGGER.isLoggable(Level.FINEST)) {
-                        LOGGER.finest("Submitting JSON: " + body);
-                    }
-
-                    try (OutputStream out = http.getOutputStream();
-                            OutputStreamWriter writer = new OutputStreamWriter(out, StandardCharsets.UTF_8)) {
-                        writer.append(body);
-                    }
-
-                    LOGGER.config("Telemetry submission received response '" + http.getResponseCode() + " " + http.getResponseMessage() + "' for: " + telemetry.getId());
-                } catch (MalformedURLException e) {
+                    URI uri = new URI(ENDPOINT);
+                    httpRequest = ProxyConfiguration.newHttpRequestBuilder(uri)
+                            .headers("Content-Type", "application/json; charset=UTF-8")
```

Comment:
Why this change? Per [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type#directives), lowercase is preferred for charset.